src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Data/GroupsVisitor.cs

```diff
@@ -0,0 +1,139 @@
+{
+	internal class GroupsVisitor : ICollectionUpdaterVisitor
+	{
+		private readonly IUpdateContext _context;
+		private readonly ILayerHolder _source;
+		private readonly FlatCollectionChangedFacet _flatCollectionChanged;
+		private readonly DataLayerCollection _children;
+
+		public GroupsVisitor(
+			IUpdateContext context,
+			ILayerHolder source,
+			FlatCollectionChangedFacet collectionChanged, 
+			DataLayerCollection children)
+		{
+			_context = context;
+			_source = source;
+			_flatCollectionChanged = collectionChanged;
+			_children = children;
+		}
+
+		public void AddItem(object item, ICollectionUpdateCallbacks callbacks)
+			// Note: As Windows 10.15063 behavior, if we are initializing the holder (_versionType == Initialize), 
+			//		 we don't have to send events for the items that are currently in the list.
+			=> AddItem(item, callbacks, raiseAddOnFlat: _context.Type != VisitorType.InitializeCollection);
+
+		private void AddItem(object item, ICollectionUpdateCallbacks callbacks, bool raiseAddOnFlat)
+		{
+			var group = (IObservableGroup)item;
+
+			// 1. Create the holder for the child, which enables the tracking of those group itself.
+			//	  ie. Enables the event 'this.View.Groups[X].CollectionChanged'.
+			// Note: Currently we support only one level of branch, so we always create a LeafHolder.
+			var (holder, initializer) = _source.CreateSubLayer(group, _context);
+			var view = new BindableCollectionGroup(group, holder);
+
+			_children.Add(group, view); // Attach the view to the model, so it will be visible to everybody through the converter
+
+			callbacks.Prepend(initializer);
+			callbacks.Append(SubscribeToChild);
+			callbacks.Append(initializer.ParentAfter); // We must allow the child to unbuffer it change only after having subscribe to it
+
+			void SubscribeToChild()
+			{
+				// 2. Once the group has been added into the view, ensure propagation of the collection changes from the child to the flat root view 
+				//    and raise add for items that are already present in the view.
+				//	  ie. Enables 'this.View.CollectionChanged(group.item)'
+				// So give 'null' instead of the actual current items.
+				var items = raiseAddOnFlat && !_context.HasReachedLimit 
+					? initializer.Result
+					: null;
+
+				_flatCollectionChanged.AddChild(holder.GetFacet<CollectionChangedFacet>(), holder.View, items);
+			}
+		}
+
+		public void SameItem(object original, object updated, ICollectionUpdateCallbacks callbacks)
+		{
+			var oldGroup = (IObservableGroup)original;
+			var newGroup = (IObservableGroup)updated;
+
+			// Even if the items are the "Same" it may be another instance, so ensure to make the _children aware of the new instance.
+			_children.Update(oldGroup, newGroup);
+		}
+
+		public bool ReplaceItem(object original, object updated, ICollectionUpdateCallbacks callbacks)
+		{
+			var oldGroup = (IObservableGroup)original;
+			var newGroup = (IObservableGroup)updated;
```

Comment:
Is this always valid?

Reply:
Yes, we are wrapping invalid group types to those at the root of the `BindableCollection` (cf. `DataStructure`).